### PR TITLE
Feature/UI

### PR DESCRIPTION
**Pull request by mephi13**
User interface

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -134,8 +137,13 @@ certs/*.csr
 # Ignore the log volume
 log/
 
+<<<<<<< HEAD
```
Comment (aCinal) [inline .gitignore:140]: Remove git conflict markers

File: client/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,14 @@
+blessed==1.19.1
```
Comment (aCinal) [inline client/requirements.txt:1]: Why is requirements.txt needed? Can't this be resolved via setup.cfg?

File: client/src/client/user_interface/__init__.py [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+                        tile.chat_with, new_message  # type: ignore
+                    )  # type: ignore
+                    # some callback to client would be needed
+                    # TODO: add callback for client
+                elif tile and tile_type == Tile:
+                    tile.consume_input(inp, self.term)
+                    pass
+
+    def say_hello(self) -> None:
+        """Says fucking hello."""
```
Comment (aCinal) [inline client/src/client/user_interface/__init__.py:138]: Watch your profanity ;)

File: client/src/client/user_interface/tiling_managers.py [hunk 4]
```diff
@@ -0,0 +1,931 @@
...
+        )
+        c.absolute_sizes = []
+        c.screen_rect = self.screen_rect
+        c.ratio = self.ratio
+        c.align = self.align
+        c.use_margins = self.use_margins
+        c.term = self.term
+        return c
+
+    def _get_relative_size_from_absolute(self, absolute_size: int) -> float:
```
Comment (aCinal) [inline client/src/client/user_interface/tiling_managers.py:205]: Missing docstring

File: client/src/client/user_interface/tiling_managers.py [hunk 5]
```diff
@@ -0,0 +1,931 @@
...
+        return absolute_size / self.screen_rect.height
+
+    def _get_absolute_size_from_relative(self, relative_size: float) -> int:
```
Comment (aCinal) [inline client/src/client/user_interface/tiling_managers.py:208]: Missing docstring

File: client/src/client/user_interface/view.py [hunk 6]
```diff
@@ -0,0 +1,302 @@
...
+            to_user=eve,
+            body="heya",
+            date=datetime(
+                year=2022, month=3, day=21, hour=11, minute=36, second=15
+            ),
+        ),
+    ]
+
+    messages_with_bob.reverse()
+    for i in range(0, len(messages_with_bob)):
```
Comment (aCinal) [inline client/src/client/user_interface/view.py:296]: Probably simpler to write for message in messages_with_bob :)

File: .gitignore (hunk 1 above)
Comment (mephi13) [reply .gitignore:140]: whoops

**Review by aCinal — APPROVED** [review]

**Pull request merged**